src/zfstools/models.py

```diff
@@ -4,18 +4,23 @@
 
 from collections import OrderedDict
 
-class Dataset:
+class Dataset(object):
     name = None
     children = None
+    _properties = None
     parent = None
     invalidated = False
     def __init__(self, name, parent=None):
         self.name = name
         self.children = []
+        self._properties = {}
         if parent:
             self.parent = parent
             self.parent.add_child(self)
 
+    def __getattr__(self,name):
```

Comment:
This looks elegant but it introduces an indirection layer that I'm uncomfortable with.  If get_creation calls were changed to get_property(creation) everywhere, and your code used get_property everywhere consistently, I'd be okay with that.